lib/ansible/modules/cloud/digital_ocean/do_sshkeys.py

```diff
@@ -0,0 +1,267 @@
+
+    if status_code not in (200, 404):
+        module.fail_json(msg='Error getting ssh key [{}: {}]'.format(
+            status_code, response.json['message']), fingerprint=fingerprint)
+
+    if state in ('present'):
+        if status_code == 404:
+            # IF key not found create it!
+
+            if module.check_mode:
+                module.exit_json(changed=True)
+
+            payload = {
+                'name': name,
+                'public_key': ssh_pub_key
+            }
+            response = rest.post('account/keys', data=payload)
+            status_code = response.status_code
+            json = response.json
+            if status_code == 201:
+                module.exit_json(changed=True, data=json)
+
+            module.fail_json(msg='Error creating ssh key [{}: {}]'.format(
+                status_code, response.json['message']))
+
+        elif status_code == 200:
+            # If key found was found, check if name needs to be updated
+            if name is None or json['ssh_key']['name'] == name:
+                module.exit_json(changed=False, data=json)
+
+            if module.check_mode:
+                module.exit_json(changed=True)
+
+            payload = {
+                'name': name,
+            }
+            response = rest.put('account/keys/{}'.format(fingerprint), data=payload)
+            status_code = response.status_code
+            json = response.json
+            if status_code == 200:
+                module.exit_json(changed=True, data=json)
+
+            module.fail_json(msg='Error updating ssh key name [{}: {}]'.format(
+                status_code, response.json['message']), fingerprint=fingerprint)
+
+    elif state in ('absent'):
+        if status_code == 404:
+            module.exit_json(changed=False)
+
+        if module.check_mode:
+            module.exit_json(changed=True)
+
+        response = rest.delete('account/keys/{}'.format(fingerprint))
+        status_code = response.status_code
+        json = response.json
+        if status_code == 204:
+            module.exit_json(changed=True)
+
+        module.fail_json(msg='Error creating ssh key [{}: {}]'.format(
+            status_code, response.json['message']))
+
+
+def ssh_key_fingerprint(ssh_pub_key):
+    key = ssh_pub_key.split(None, 2)[1]
+    fingerprint = hashlib.md5(base64.decodestring(key)).hexdigest()
+    return ':'.join(a + b for a, b in zip(fingerprint[::2], fingerprint[1::2]))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            fingerprint=dict(aliases=['id'], required=False),
+            name=dict(required=False),
+            ssh_pub_key=dict(required=False),
+            oauth_token=dict(
+                no_log=True,
+                # Support environment variable for DigitalOcean OAuth Token
+                fallback=(env_fallback, ['DO_API_TOKEN', 'DO_API_KEY']),
```

Comment:
After our discussion on PR #18922 I think you should add support for oauth_token in this module.